### DISCUSSION TITLE
Drag and Drop image from browser using touch

### DISCUSSION THREAD

**Discussion by smd-11** [cefsharp/CefSharp#3630] [General]
i am working on wpf application. is there any way i can drag image from browser to canvas using touch.i am able to do that using mouse but not by touch

**Comment by amaitland** [comment]
Make sure you've using the newer windows touch implementation see https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/mitigation-pointer-based-touch-and-stylus-support

Past that I don't have a touch screen at the moment so you'll have to do the debugging. 

Add a breakpoint to https://github.com/cefsharp/CefSharp/blob/cefsharp/91/CefSharp.Wpf/ChromiumWebBrowser.cs#L858 for starters, see if the drag start is called.

**Reply by amaitland** [reply]
Alternatively you can try switching to  https://www.nuget.org/packages/CefSharp.Wpf.HwndHost/ which is the equivalent of using the WinForms version in WPF using a WindowsFormsHost. Standard airspace issues apply.

**Reply by smd-11** [reply]
Drag start is not called when using touch,checked in cefsharp.wpf.example

**Reply by amaitland** [reply]
This maybe a bug in the CEF implementation, you'll need to find a way to test with the CEF Sample application. 

Details for testing at https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef
